### PR TITLE
 emit signal and actions events in correct order

### LINKED ISSUE #176: some async actions do not emit `actionEnd` event

### PR DESCRIPTION
**Pull request by Guria**
resolves #176 
@christianalfoni review process:
```sh
git checkout issue-176
git checkout HEAD~1 # switch to commit with failing tests
npm test # examine failing test
git checkout issue-176 # switch to fix
npm test # examine successful test
```
* say if it looks good for you in comment
* avoid using green `Merge pull-request` button
* I will push it to master

### CODE REVIEW

**Comment by christianalfoni** [comment]
That worked :-) So the idea now is that I always create a branch. That branch should be reviewed by someone before pulled into master? Any bug fixes are named "issue-xxx" and other branches are considered feature branches?

**Comment by Guria** [comment]
@christianalfoni yes for all. always a branch and review by someone (I suppose it could be omitted in very simple cases, like docs, fixing typos, etc).

**Comment by Guria** [comment]
@christianalfoni what abou this pr?

File: src/CreateSignalFactory.js [hunk 1]
```diff
@@ -184,6 +184,8 @@ module.exports = function (signalStore, recorder, devtools, controller, model, s
                   action.output = result.arg
                   utils.merge(signalArgs, result.arg)
                   signalStore.removeAsyncAction(action)
+                  controller.emit('actionEnd', {action: action, signal: signal})
+                  controller.emit('change', {signal: signal})
```
Comment (Guria) [inline src/CreateSignalFactory.js]: do we ever need to emit `change` after async `actionEnd`?

**Comment by christianalfoni** [comment]
Looks good to me :-)

**Pull request merged**